openedx_tagging/core/tagging/rest_api/v1/views_import.py

```diff
@@ -49,3 +52,47 @@ def get(self, request: Request, file_ext: str, *args, **kwargs) -> FileResponse:
         response = FileResponse(fh, content_type=content_type)
         response['Content-Disposition'] = content_disposition
         return response
+
+
+class ImportView(APIView):
+    """
+    View to import taxonomies
+
+    **Example Requests**
+        POST /tagging/rest_api/v1/import/
+        {
+            "taxonomy_name": "Taxonomy Name",
+            "taxonomy_description": "This is a description",
+            "file": <file>,
+        }
+
+    **Query Returns**
+        * 200 - Success
+        * 400 - Bad request
+        * 405 - Method not allowed
+    """
+    http_method_names = ['post']
+
+    def post(self, request: Request, *args, **kwargs) -> HttpResponse:
+        """
+        Imports the taxonomy from the uploaded file.
+        """
+        body = TaxonomyImportBodySerializer(data=request.data)
+        body.is_valid(raise_exception=True)
+
+        taxonomy_name = body.validated_data["taxonomy_name"]
+        taxonomy_description = body.validated_data["taxonomy_description"]
+        file = body.validated_data["file"].file
+        parser_format = body.validated_data["parser_format"]
+
+        import_success = api.create_taxonomy_and_import_tags(
+            taxonomy_name=taxonomy_name,
+            taxonomy_description=taxonomy_description,
+            file=file,
+            parser_format=parser_format,
+        )
```

Comment:
Hmm.. the import/export tags API was designed to [use async celery tasks](https://github.com/openedx/openedx-learning/blob/05105bcc9e733619650c5041caa91215d9fa36d0/openedx_tagging/core/tagging/import_export/tasks.py) so it could run in the background (in case there's a lot of tags to import).

@bradenmacdonald do you want us to use these async tasks here, or are we doing all imports synchronously?

Reply:
I suspect synchronous will be totally fine for the short term, but we might as well do it async since we've done most of the work already. How much work will that add to make the REST API layer async too?

Reply:
I meant (1). But with simple polling for now, no need for websockets or anything fancy. For (2) it actually ties up much _more_ server resources than just directly doing the import synchronously.

If that's going to take a while to implement though we can just to sync for now.

Reply:
Got it!
There is only one problem in our flow:
We need to create the Taxonomy before calling the import.

If the import fails, it is easy to remove the created taxonomy in the sync call.

Doing async, we can create the taxonomy, return the id to the client, and make calls waiting for the results (the result is associated with the taxonomy). But I need to figure out how to delete the freshly created taxonomy after a failed attempt. Would you happen to have any ideas on this?

This is not urgent for this task, but we will need to handle it if we want to go async sometime in the future. One approach is to change the import flow in the front-end: we first create the taxonomy and always call import inside it (then we don't need to handle the delete). This will impact our designs and this task a bit.

Another approach is to handle the create+import sync and use async for importing on an already created taxonomy. But it will also impact the UX, having different flows.

Reply:
I think the right approach is to create an additional async task which creates the taxonomy, then synchronously calls the existing import task (which is a celery task so can be called either sync or async, but since we're already in an async task, we call it sync). If that succeeds, it returns the taxonomy ID, and if it fails, it deletes the taxonomy and returns an error message.

The main thing is that this "wrapper" task which does create+import returns an _import ID_ not a taxonomy ID. And then the frontend polls the import ID until it gets either a success or failure message. Only then does it get the taxonomy ID.

--------

But that's all getting too complicated. Just do it synchronously for now an we'll make it async if it's too slow in practice. Explicitly put a comment in the REST API for the import that "this is an unstable API and may change if we make it async."